### PR TITLE
Improvements for provisioner scripts

### PR DESCRIPTION
**Pull request by szkl**

### CODE REVIEW

File: scripts/000_install_base.sh [hunk 1]
```diff
@@ -10,6 +10,9 @@ pushd $(dirname $0)
 ./040_install_go.sh
 ./050_install_python.sh
 ./050_install_supervisor.sh
+./051_install_mongodb.sh
```
Comment (cihangir) [inline scripts/000_install_base.sh]: We dont need these in base. You can add them to the json file for the specific purpose.

File: scripts/001_sysctl.sh [hunk 2]
```diff
@@ -0,0 +1,29 @@
+#!/bin/bash
```
Comment (cihangir) [inline scripts/001_sysctl.sh]: 👍

File: scripts/900_cleanup.sh [hunk 3]
```diff
@@ -1,5 +1,9 @@
 #!/bin/bash
 
-yum clean all && rm -rf /tmp/* /var/tmp/*
+yum clean all
+rm -rf /tmp/* /var/tmp/*
 
-grep -v $HOSTNAME /etc/hosts >/tmp/hosts && mv --force /tmp/hosts /etc/hosts
+if command -v hostname; then
+	sed --expression "/$(hostname)/ d" /etc/hosts >/tmp/hosts
```
Comment (cihangir) [inline scripts/900_cleanup.sh]: I am not sure how this supposed to work with lowercase env var. Does sed support auto replacing of hostname?

On the other the reasoning behind these lines were to clean the hostname if it exists to prepare for the 
actual usage of the AMI.

I don't see a necessity to check if hostname command exists or not

**Review by cihangir — CHANGES_REQUESTED** [review]

File: scripts/000_install_base.sh (hunk 1 above)
Comment (szkl) [reply scripts/000_install_base.sh]: They are needed for development purposes. That's why I thought it'd be better to have them in the base considering container images to be used for both development and deployment.

File: scripts/900_cleanup.sh (hunk 3 above)
Comment (szkl) [reply scripts/900_cleanup.sh]: > I am not sure how this supposed to work with lowercase env var. Does sed support auto replacing of hostname?

It executes hostname program, not a variable substitution. sed expression is simply deleting any lines that contain the hostname.

> I don't see a necessity to check if hostname command exists or not

It doesn't exist in `amazonlinux` docker image.

Comment (cihangir) [reply scripts/900_cleanup.sh]: > It executes hostname program, not a variable substitution.

My bad, sorry :)

**Review by cihangir — APPROVED** [review]
we are talking for a possible solution on chat, approving not to require another sync

**Pull request merged**